base/reflection.jl

```diff
@@ -1204,22 +1211,22 @@ julia> hasmethod(g, Tuple{}, (:a, :b, :c, :d))  # g accepts arbitrary kwargs
 true
 ```
 """
-function hasmethod(@nospecialize(f), @nospecialize(t); world = typemax(UInt))
+function hasmethod(@nospecialize(f), @nospecialize(t); world=typemax(UInt))
     t = to_tuple_type(t)
     t = signature_type(f, t)
     return ccall(:jl_method_exists, Cint, (Any, Any, UInt), typeof(f).name.mt, t, world) != 0
 end
 
 function hasmethod(@nospecialize(f), @nospecialize(t), kwnames::Tuple{Vararg{Symbol}}; world=typemax(UInt))
+    # TODO: this appears to be doing the wrong queries
```

Comment:
Can you open an issue for this? I added this method so I'd be interested to know what I did wrong.

Reply:
I don't think the right queries are exposed. This is close enough, and is perhaps a better approach than what we actually do currently.